<h2>Heat Exchange System of Diffeqs Undetermined Coefficients</h2>

<font color="green">Schmierbuch 3640</font>

The accompanying illustration is a
cutaway view of a two-room cabin. The
temperatures in the Outer and Inner
room, respectively, at time $t$ are given
by $A(t)$ and $B(t)$. A space heater in the
Inner Room gives off sufficient heat to
raise the temperature of a fully insulated
room by $H=3.0^{\circ}C$ per hour. Both
rooms, however, continually lose heat to
the outside and to each other. Moreover,
the outside temperature $E$ (measured in
degrees Celsius) varies sinusoidally with
the time of day according to the function:
$$
E=4+9\cos\frac{\pi{}t}{12}t
$$
where $t$ is measured in hours.
The temperature in each room can be
modeled, using Newton’s Law of
Cooling, as a system of two coupled
differential equations:
$$
\begin{align*}
\frac{dA}{dt}&=k_3(E-A)+k_2(B-A)\\
\frac{dB}{dt}&=k_1(E-B)-k_2(B-A)+H
\end{align*}
$$
where $k_1=0.5$ parameterizes the rate at which room B loses heat to the outside (primarily
through the window), $k_2=0.7$ parameterizes the rate at which heat is transferred between
the rooms (primarily via the connecting door), and $k_3=0.4$ parameterizes the rate at which
room A loses heat to the outside (primarily through the window and the outside door).

* Determine the complementary solution for the given system of equations.
* Determine the particular solution for the given system of equations using the method of
undetermined coefficients.
* Assuming that the Outer Room has an initial temperature of $A(0)=3^{\circ}C$ and that the initial
temperature in the Inner Room is $B(0)=4^{\circ}C$, find the temperatures of each room as functions
of time. Plot the temperature of both rooms and the temperature of the environment together
on a single plot showing the temperature variations over the first 48 hours.

For Term Test B:
* Determine the complementary solution for the given system of equations.
* Construct a trial solution in order to find a particular solution. You do not need to find this particular solution, just the trial solution with undetermined coefficients.

In [30]:
A1=[-1.1 0.7
0.7 -1.2]

2×2 Matrix{Float64}:
 -1.1   0.7
  0.7  -1.2

In [31]:
using LinearAlgebra

In [33]:
eva=eigvals(A1)

2-element Vector{Float64}:
 -1.8517834423809099
 -0.44821655761909

In [34]:
eve=eigvecs(A1)

2×2 Matrix{Float64}:
 -0.681452  -0.731863
  0.731863  -0.681452

In [8]:
aa=pi/12

0.2617993877991494

In [35]:
M1=[0 0 aa 0 1.1 -0.7
0 0 1.1 -0.7 -aa 0
1.1 -0.7 0 0 0 0
0 0 0 aa -0.7 1.2
0 0 -0.7 1.2 0 -aa
-0.7 1.2 0 0 0 0]

6×6 Matrix{Float64}:
  0.0   0.0   0.261799   0.0        1.1       -0.7
  0.0   0.0   1.1       -0.7       -0.261799   0.0
  1.1  -0.7   0.0        0.0        0.0        0.0
  0.0   0.0   0.0        0.261799  -0.7        1.2
  0.0   0.0  -0.7        1.2        0.0       -0.261799
 -0.7   1.2   0.0        0.0        0.0        0.0

In [36]:
v1=[3.6;0;1.6;4.5;0;5]

6-element Vector{Float64}:
 3.6
 0.0
 1.6
 4.5
 0.0
 5.0

In [37]:
Cf=inv(M1)*v1

6-element Vector{Float64}:
 6.53012048192771
 7.975903614457832
 4.011396165515925
 3.821013940359336
 6.63800644617714
 6.788554501638257

The solution is
$$
x(t)=\left[\begin{array}{c}6.530\\7.976\end{array}\right]+\left[\begin{array}{c}4.011\\3.821\end{array}\right]\sin\frac{\pi{}t}{12}+\left[\begin{array}{c}6.638\\6.789\end{array}\right]\cos\frac{\pi{}t}{12}+
$$
$$
c_1\cdot\left[\begin{array}{c}-0.681\\0.732\end{array}\right]e^{-1.852t}+
c_2\cdot\left[\begin{array}{c}-0.732\\-0.681\end{array}\right]e^{-0.448t}
$$

In [40]:
cc=inv(eve)*[3-Cf[1]-Cf[5];4-Cf[2]-Cf[6]] # c1 and c2

2-element Vector{Float64}:
 -0.9490213604978094
 14.777135115324516

<img src="./diagrams/vytdnmpe.png" width="70%" height="70%">